chain/chain/src/runtime/mod.rs

```diff
@@ -650,16 +651,16 @@ impl RuntimeAdapter for NightshadeRuntime {
         if ProtocolFeature::StatelessValidation.enabled(next_protocol_version)
             || cfg!(feature = "shadow_chunk_validation")
         {
-            trie = trie.recording_reads_new_recorder();
+            let proof_size_limit =
+                runtime_config.witness_config.new_transactions_validation_state_size_soft_limit;
+            trie = trie.recording_reads_with_proof_size_limit(proof_size_limit);
```

Comment:
I'm a bit surprised that there isn't a matching removed line with this config.

Reply:
Are you referring to line 2001 in runtime/runtime/src/lib.rs?